Sources/AST/Environment/Environment+Memory.swift

```diff
@@ -21,10 +21,6 @@ extension Environment {
     case .errorType: return 0
     case .functionType: return 0
 
-    case .stdlibType(let type):
```

Comment:
I'm not actually 100% on what this did, we may need to add handling for value Wei types. However it seems a bit daft to only do this for Wei, so perhaps this alone = passing by value for structure types?

Reply:
Is this tested for? If it is passed by value then the value in that asset is copied, so modifying it in the called function doesn't change it in the caller function? That would definitely be wrong.

I would say that this is just an error resulting from the `implicit` stuff. Within Flint code, `Wei` should 100% be passed around like any other `struct`, i.e. by reference.

Reply:
Makes sense, and I wholly agree.
https://github.com/flintrocks/flint/blob/master/Tests/SemanticTests/fallback.flint
and a few other tests would need to be changed since we currently allow this. So rather than allowing Wei to be passed by value, it may only be 'passed by value' if part of an implicit param.

That means I would need to change https://github.com/flintrocks/flint/pull/56/files#diff-e42b8f968d8f53fdde59f432558adcafR147 to allow implicit currencies rather than any currency. I'll attach a don't merge label for now until I have gotten around to doing this.